### ISSUE TITLE
sysv init script not compatible with * in LS_CONF_DIR paths

### ISSUE THREAD

**Issue by robin13** [elastic/logstash#4736]
If you have e.g. `LS_CONF_DIR=/etc/logstash/conf.d/*/*.conf` which would be useful for a configuration structure like this:

```
conf.d/1-inputs/file.conf
conf.d/1-inputs/tcp.conf
conf.d/1-inputs/beats.conf
conf.d/9-outputs/elasticsearch.conf
conf.d/9-outputs/file.conf
```

Then `service logstash restart` fails because [L139](https://github.com/elastic/logstash/blob/master/pkg/logstash.sysv#L139) looks for ` ${LS_CONF_DIR}/*`

Perhaps better first identify if the path already contains globs, and is a directory (not a single file), and only then append the `/*`?

**Comment by jordansissel** [comment]
the variable includes `DIR` which makes me think it's a directory, so having it support wildcards or paths seems unexpected given the name. I am open to supporting what you propose, but I think we need to rename the variable given the meaning will change.

What about renaming it to LS_CONF_PATH ?

Looking towards the future, I would _highly_ recommend you move forward towards more modern service management systems (upstart, systemd, etc) as we will likely be moving away from the older SYSV init script style things eventually.

**Comment by jordansissel** [comment]
(reminder to myself to file a separate ticket to move our rpm/deb packages to systemd)

**Comment by untergeek** [comment]
This needs to be re-evaluated by virtue of the fact that LS_CONF_PATH is no longer even used, due to the shift to using `logstash.yml` as the settings path.  While Logstash _may_ yet need to look at directories and/or sub-directories, this is a different world with `path.settings`.

Also, the SYSV issue has been completely resolved by #5341, which changes startup altogether.

If this is still an issue, please open it again after testing against 5.0 alpha-3, with regards to the `path.settings` setup.  Feel free to reference this issue in that new issue.
